libpromises/eval_context.c

```diff
@@ -1443,6 +1450,8 @@ static bool EvalContextClassPut(EvalContext *ctx, const char *ns, const char *na
         return false;
     }
 
+    StringSetAdd(ctx->all_classes, SafeStringDuplicate(name));
```

Comment:
The thing is that masterfiles is only basic policy. We know (from the recent class collision issue) that customers have way more complex policies, setting hundreds of thousands of classes during the agent lifetime.